core/modules/field_ui/field_ui.admin.inc

```diff
@@ -938,6 +938,13 @@ function field_ui_display_form($form, &$form_state, $entity_type, $bundle, $view
   elseif (is_object($bundle) && property_exists($bundle, 'label')) {
     $bundle_name = check_plain($bundle->label);
   }
+  elseif (is_string($bundle)) {
+    $entity_info = entity_get_info($entity_type);
+    if (isset($entity_info['label'])) {
+      $bundle_name = $entity_info['label'];
```

Comment:
The bundle name above is sanitized, but this one is not. I believe we should run this through check_plain():
```suggestion
      $bundle_name = check_plain($entity_info['label']);
```

Reply:
However, down below the only place this is used is to set the title in `backdrop_set_title()`, which takes the 2nd argument to filter the string. We could either:

A) Remove check_plain() from the label in both places.
B) Set the 2nd argument in backdrop_set_title() to `PASS_THROUGH` to prevent double-escaping.

Reply:
Option B (PASS_THROUGH) seems problematic to me, because field_ui_manage_display_title() returns the raw `$view_mode['label']` - if the view mode isn't default.

I'd prefer to remove the redundant check_plain() from the $bundle->name and $bundle->label and let backdrop_set_title() do the sanitize task.

Reply:
I would expect that the ideal thing would be that anything that comes out of `$bundle->label` is already passed through `check_plain()` ...but my knowledge of how the internals of the entity system works is very limited.